forc-plugins/forc-doc/src/descriptor.rs

```diff
@@ -54,29 +54,45 @@ pub(crate) enum Descriptor {
 }
 
 impl Descriptor {
-    pub(crate) fn from_typed_decl(d: &TyDeclaration, module_prefix: Vec<String>) -> Result<Self> {
+    pub(crate) fn from_typed_decl(
+        d: &TyDeclaration,
+        module_prefix: Vec<String>,
+        document_private_items: bool,
+    ) -> Result<Self> {
         use TyDeclaration::*;
         match d {
             StructDeclaration(ref decl_id) => {
                 let struct_decl = de_get_struct(decl_id.clone(), &decl_id.span())?;
-                Ok(Descriptor::Documentable {
-                    module_prefix,
-                    desc_ty: Box::new(DescriptorType::Struct(struct_decl)),
-                })
+                if !document_private_items && struct_decl.visibility.is_private() {
```

Comment:
So this is based off of whether the `--document-private-items` tag is passed to `forc doc`. This reads "if document private items wasn't passed and the declaration is private then this declaration is `NonDocumentable`. otherwise, document it". Maybe there is a way to write it with logical `or`, would you mind clarifying? Not all declaration types have a visibility field, so there is also that to take into account.

Reply:
The way this read to me is "if you do not want to document private items AND the item is private then do not document it" meaning that if `assert!(document_private_items)` but it's private then you document it anyway

Reply:
You can flip this and say:
```
if document_private_items || !struct_decl.visibility.is_private() {
    Ok (Descriptor::Documentable {..})
} else {
    Ok (Descriptor::NonDocumentable)
}
```
But this seems similar in terms of mental overhead. In fact, I think like the `&&` version better. It says: don't document private structs if `document_private_items` is `false`.

Reply:
Well we don't want an or here I think because we need to check that the user wants to document private items, and if they don't then we also want to know if any items are private so they are both bools need to be evaluated together

Reply:
there's also an `is_public()` method haha